File: src/main/java/com/itsci/mju/maebanjumpen/entity/HousekeeperSkill.java
package com.itsci.mju.maebanjumpen.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Entity
@Table(name = "housekeeper_skill")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
public class HousekeeperSkill {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Integer skillId;

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "skill_level_tier_id", nullable = false)
        @ToString.Exclude
        private SkillLevelTier skillLevelTier;

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "housekeeper_id", nullable = false)
        @JsonBackReference("housekeeper-skills")
        @ToString.Exclude
        private Housekeeper housekeeper;

        @ManyToOne(fetch = FetchType.EAGER)
        @JoinColumn(name = "skill_type_id", nullable = false)
        @ToString.Exclude
        private SkillType skillType;

        @Column(name = "price_per_day")
        private Double pricePerDay;

        @Column(name = "total_hires_completed", nullable = false)
        private Integer totalHiresCompleted = 0;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/HousekeeperSkill.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*

@Entity
@Table(name = "housekeeper_skill")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class HousekeeperSkill(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "skill_level_tier_id", nullable = false)
    var skillLevelTier: SkillLevelTier? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "housekeeper_id", nullable = false)
    @JsonBackReference("housekeeper-skills")
    var housekeeper: Housekeeper? = null,

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "skill_type_id", nullable = false)
    var skillType: SkillType? = null,

    @Column(name = "price_per_day")
    var pricePerDay: Double? = null,

    @Column(name = "total_hires_completed", nullable = false)
    var totalHiresCompleted: Int = 0
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is HousekeeperSkill) return false
        return id != null && id == other.id
    }

    override fun hashCode(): Int = id?.hashCode() ?: 0

    override fun toString(): String = "HousekeeperSkill(skillId=$id, pricePerDay=$pricePerDay)"
}

